cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -210,14 +211,29 @@ void reader::impl::decode_page_data(read_mode mode, size_t skip_rows, size_t num
     }
   }
 
+  // A hostdevice vector to compute and store initial str offsets from decoders to build nested
+  // large string columns. For non-nested large strign columns, the initial string offset is always
+  // zero.
+  auto initial_str_offsets =
+    cudf::detail::hostdevice_vector<size_t>{_input_columns.size(), _stream};
```

Comment:
IMO it would be better to use a cudf::detail::host_vector + make_device_uvector than to use a `hostdevice_vector`, given that the data isn't copied back to host (in this case the host vector can be declared in `if (has_strings) {` branch). Might be even better to directly initialize the device memory.

Reply:
The data is actually copied back to the host at L421. Though I agree that if `has_strings` is not true, we don't really need this.

Reply:
Replaced with explicit host and device vectors. Can't use `thrust::fill` from within the .cpp file to directly initialize the device memory here but can create a util function in a .cu file to do this.